overlord/snapstate/refreshhints.go

```diff
@@ -121,3 +136,52 @@ func (r *refreshHints) Ensure() error {
 	}
 	return r.refresh()
 }
+
+func refreshHintsFromCandidates(st *state.State, updates []*snap.Info, ignoreValidationByInstanceName map[string]bool, deviceCtx DeviceContext) ([]*refreshCandidate, error) {
+	if ValidateRefreshes != nil && len(updates) != 0 {
+		userID := 0
+		var err error
+		updates, err = ValidateRefreshes(st, updates, ignoreValidationByInstanceName, userID, deviceCtx)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	hints := []*refreshCandidate{}
+	for _, update := range updates {
+		var snapst SnapState
+		if err := Get(st, update.InstanceName(), &snapst); err != nil {
+			return nil, err
+		}
+
+		flags := snapst.Flags
+		flags.IsAutoRefresh = true
+		flags, err := earlyChecks(st, &snapst, update, flags)
+		if err != nil {
+			logger.Noticef("cannot update %q: %v", update.InstanceName(), err)
```

Comment:
we weren't producing  these logs before from refreshHints, right? we probably want to log different things for refreshHints and the main auto-refresh later?  we could pass this a logging callback?  or for now just turn these into Debugf with messages like and leave a TODO about the other callers to this?

"update hint for %q is not applicable: %v"

Reply:
Done.